handler.go

```diff
@@ -0,0 +1,269 @@
+
+		default:
+			lookupErr := fmt.Errorf("lookup type %T not known", x)
+			handleError(ctx, lookupErr, svcerror.Code_BAD_REQUEST, reply, conn)
+			return
+		}
+
+		selectFields := []interface{}{
+			usersT.Col("id"),
+			usersT.Col("username"),
+		}
+
+		if request.IncludeSavedSearches {
+			selectFields = append(
+				selectFields,
+				savedSearchesT.Col("id").As("saved_searches_id"),
+				savedSearchesT.Col("saved_searches").As("saved_searches"),
+			)
+		}
+
+		if request.IncludePreferences {
+			selectFields = append(
+				selectFields,
+				prefsT.Col("id").As("preferences_id"),
+				prefsT.Col("preferences").As("preferences"),
+			)
+		}
+
+		q = q.Select(selectFields...)
+
+		queryString, _, err := q.ToSQL()
+		if err != nil {
+			handleError(ctx, err, svcerror.Code_INTERNAL, reply, conn)
+		}
+
+		log.Infof("%s\n", queryString)
+
+		u := lookupUser{}
+
+		// argument handling is done by the goqu library above.
+		if err = dbconn.QueryRowxContext(context.Background(), queryString).StructScan(&u); err != nil {
+			handleError(ctx, err, svcerror.Code_INTERNAL, reply, conn)
+		}
+
+		responseUser := user.User{
+			Uuid:     u.UserID,
+			Username: u.Username,
+			Preferences: &user.User_Preferences{
+				Uuid:        u.PreferencesID,
+				Preferences: u.Preferences,
+			},
+			SavedSearches: &user.User_SavedSearches{
+				Uuid:          u.SavedSearchesID,
+				SavedSearches: u.SavedSearches,
+			},
+		}
+
+		if request.IncludeLogins {
+			logins, err := lookupLogins(
+				dbconn,
+				u.UserID,
+				uint(request.LoginLimit),
+				uint(request.LoginOffset),
+			)
+			if err != nil {
+				handleError(ctx, err, svcerror.Code_INTERNAL, reply, conn)
+			}
+
+			responseUser.Logins = []*user.User_Login{}
+
+			for _, login := range logins {
+				ul := user.User_Login{}
+				if login.IPAddress.Valid {
+					ul.IpAddress = login.IPAddress.String
+				}
+				if login.UserAgent.Valid {
+					ul.UserAgent = login.UserAgent.String
+				}
+				if login.LoginTime.Valid {
+					ul.LoginTime = timestamppb.New(login.LoginTime.Time)
+				}
+				if login.LogoutTime.Valid {
+					ul.LogoutTime = timestamppb.New(login.LogoutTime.Time)
+				}
+				responseUser.Logins = append(responseUser.Logins, &ul)
+			}
+
+			loginCount, err := loginCount(dbconn, u.UserID)
+			if err != nil {
+				handleError(ctx, err, svcerror.Code_INTERNAL, reply, conn)
+			}
+
+			responseUser.LoginCount = uint32(loginCount)
+		}
+
+		if err = publishResponse(ctx, conn, reply, &responseUser); err != nil {
+			log.Error(err)
+		}
+	}
+}
+
+func publishResponse(ctx context.Context, conn *nats.EncodedConn, reply string, responseUser *user.User) error {
+	carrier := gotelnats.PBTextMapCarrier{
+		Header: responseUser.Header,
+	}
+
+	_, span := gotelnats.InjectSpan(ctx, &carrier, reply, gotelnats.Send)
+	defer span.End()
+
+	return conn.Publish(reply, &responseUser)
+}
+
+func handleError(ctx context.Context, err error, code svcerror.Code, reply string, conn *nats.EncodedConn) {
```

Comment:
Since we've got a whole nice error handling function, it'd be good to mark the span it's in as erroring.

To do that, if I've got it right:

```go
// import "go.opentelemetry.io/otel/codes" & "go.opentelemetry.io/otel/trace" if they aren't
span := trace.SpanFromContext(ctx) // or pass the span into handleError
span.RecordError(err)
span.SetStatus(codes.Error, err.Error())
```

Reply:
This should be fixed, thanks.